### PR TITLE
Add Referral Task Card component

### LINKED ISSUE #95265: [2 pts left] Create Task Card Component
Image
 - [ ] Create component for displaying referral as task card
 - [ ] component takes referral object as prop
 - [ ] fetch referral from redux by UUID
 - [ ] conditionally show if UUID passed and feature is on
 - [ ] show date in `Month DD, YYYY` format
 - [ ] add unit tests

### PR DESCRIPTION
**Pull request by juancstlm-a6**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**:
   1. First search `vets-website` for _all_ instances of the `entryName` in your `manifest.json` and remove them in a separate PR. Look particularly for references in `src/applications/static-pages/static-pages-entry.js` and `src/platform/forms/constants.js`. _**If you do not do this, other applications will break!**_
      - _Add the link to your merged vets-website PR here_
   2. Then, Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
      - _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
- Adds a `Referral Task Card` component that only shows when a feature flag is on and UUID is passed to the main appointments page ex: `/appointments/?id=add2f0f4-a1ea-4dea-a504-a54ab57c68`

## Related issue(s)

department-of-veterans-affairs/va.gov-team#95265

## Testing done

- Unit tests for component logic
- Added tests for main appointments page
- In order to test this the feature `vaOnlineSchedulingCCDirectScheduling` must be enabled and a id supplied such as `/my-health/appointments/?id=add2f0f4-a1ea-4dea-a504-a54ab57c68`
- Pass in referral object and ensure the component renders as expected
- Pass in expired referral object and ensure the card does not show

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  | Screen Shot 2024-11-13 at 23 18 44|Screen Shot 2024-11-13 at 22 58 54|
| Desktop | Screenshot 2024-11-13 at 23-18-38 Appointments Veterans Affairs| Screenshot 2024-11-13 at 22-59-45 Appointments Veterans Affairs |

## What areas of the site does it impact?

- Appointments list feature in VAOS

## Acceptance criteria

 - [X] Create component for displaying referral as task card
 - [X] component takes referral object as prop
 ~~- [ ] fetch referral from redux by UUID~~
 - [X] conditionally show if UUID passed and feature is on
 - [X] show date in `Month DD, YYYY` format
 - [X] add unit tests

### Quality Assurance & Testing

- [X] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [X] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [X] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [X] Browser console contains no warnings or errors.
- [X] Events are being sent to the appropriate logging solution
- [X] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [X] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

- Data pulled from Redux will be done in separate ticket

### CODE REVIEW

File: src/applications/vaos/referral-appointments/components/ReferralTaskCard.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          numberOfAppointments,
+        )} with a community care provider. You must schedule all appointments for this referral by ${format(
+          expirationDate,
+          'PP',
+        )}.`}
+      </p>
+      <va-link-action
+        text="Go to your referral details to start scheduling"
+        type="secondary"
+        href={`/my-health/appointments/schedule-referral/${UUID}`}
```
Comment (brianseek) [inline src/applications/vaos/referral-appointments/components/ReferralTaskCard.jsx:41]: this should route via an onClick using the `routeToNextReferralPage` function that John made in the referral-appointments/flow file

**Review by brianseek — CHANGES_REQUESTED** [review]
Nice work on this one. I appreciate the tests! One little thing on routing. Also lets have a little FE CC sync up tomorrow, I think you may be able to utilize some of the mock stuff I have been working on. If not here, in a follow up ticket.

**Review by brianseek — APPROVED** [review]
LGTM

**Review by JunTaoLuo — APPROVED** [review]
Looks good on a sanity check perspective! I'll defer to Brian's approval for acknowledging the design/implementation details.

**Pull request merged**